File: web/src/models/trending.ts
import type { InstrumentType, Signal } from "./enums";

export interface TrendingItem {
  id: string;
  symbol: string;
  name: string;
  type: InstrumentType;
  price?: number | null;
  change_pct?: number | null;
  sentiment: Signal;
  trend_score: number;
  summary: string;
}

export interface TrendingResponse {
  category: string;
  items: TrendingItem[];
}

export const TRENDING_CATEGORIES = [
  { key: "trending", label: "Trending now" },
  { key: "most_bought", label: "Most bought" },
  { key: "most_sold", label: "Most sold" },
  { key: "high_momentum", label: "High momentum" },
  { key: "high_opportunity", label: "High opportunity" },
] as const;

File: backend/app/schemas/trending.py
from __future__ import annotations

from pydantic import BaseModel

from app.domain.enums import InstrumentType, Signal


class TrendingItem(BaseModel):
    id: str
    symbol: str
    name: str
    type: InstrumentType
    price: float | None = None
    change_pct: float | None = None
    sentiment: Signal = Signal.NEUTRAL
    trend_score: float = 50.0
    summary: str = ""


class TrendingResponse(BaseModel):
    category: str
    items: list[TrendingItem]
